credentials/src/main/kotlin/web5/sdk/credentials/VerifiablePresentation.kt

```diff
@@ -0,0 +1,271 @@
+ * work with JWT representations of verifiable presentations and ensuring that the signatures
+ * and claims within those JWTs can be validated.
+ *
+ * @property vpDataModel The [vpDataModel] instance representing the core data model of a verifiable presentation.
+ */
+public class VerifiablePresentation internal constructor(public val vpDataModel: VpDataModel) {
+
+  public val verifiableCredential: List<String>
+    get() = vpDataModel.toMap().get("verifiableCredential") as List<String>
+
+    public val holder: String
+    get() = vpDataModel.holder.toString()
+
+  /**
+   * Sign a verifiable presentation using a specified decentralized identifier ([did]) with the private key that pairs
+   * with the public key identified by [assertionMethodId].
+   *
+   * If the [assertionMethodId] is null, the function will attempt to use the first available verification method from
+   * the [did]. The result is a String in a JWT format.
+   *
+   * @param did The [Did] used to sign the credential.
+   * @param assertionMethodId An optional identifier for the assertion method that will be used for verification of the
+   *        produced signature.
+   * @return The JWT representing the signed verifiable credential.
+   *
+   * Example:
+   * ```
+   * val signedVp = verifiablePresentation.sign(myDid)
+   * ```
+   */
+  @JvmOverloads
+  public fun sign(did: Did, assertionMethodId: String? = null): String {
+    val didResolutionResult = DidResolvers.resolve(did.uri)
+    val assertionMethod: VerificationMethod = didResolutionResult.didDocument.findAssertionMethodById(assertionMethodId)
+
+    // TODO: ensure that publicKeyJwk is not null
+    val publicKeyJwk = JWK.parse(assertionMethod.publicKeyJwk)
+    val keyAlias = did.keyManager.getDeterministicAlias(publicKeyJwk)
+
+    // TODO: figure out how to make more reliable since algorithm is technically not a required property of a JWK
+    val algorithm = publicKeyJwk.algorithm
+    val jwsAlgorithm = JWSAlgorithm.parse(algorithm.toString())
+
+    val kid = when (assertionMethod.id.isAbsolute) {
+      true -> assertionMethod.id.toString()
+      false -> "${did.uri}${assertionMethod.id}"
+    }
+
+    val jwtHeader = JWSHeader.Builder(jwsAlgorithm)
+      .type(JOSEObjectType.JWT)
+      .keyID(kid)
+      .build()
+
+    val jwtPayload = JWTClaimsSet.Builder()
+      .issuer(did.uri)
+      .issueTime(Date())
+      .claim("vp", vpDataModel.toMap())
+      .build()
+
+    val jwtObject = SignedJWT(jwtHeader, jwtPayload)
+    val toSign = jwtObject.signingInput
+    val signatureBytes = did.keyManager.sign(keyAlias, toSign)
+
+    val base64UrlEncodedHeader = jwtHeader.toBase64URL()
+    val base64UrlEncodedPayload = jwtPayload.toPayload().toBase64URL()
+    val base64UrlEncodedSignature = Base64URL(Convert(signatureBytes).toBase64Url(padding = false))
+
+    return "$base64UrlEncodedHeader.$base64UrlEncodedPayload.$base64UrlEncodedSignature"
```

Comment:
Would it make sense to refactor  this code so it's shared with the VC.sign function?

Reply:
yup! added